packages/nimble-components/src/table/components/cell/index.ts

```diff
@@ -56,6 +67,16 @@ export class TableCell<
         }
     }
 
+    public onActionMenuBeforeToggle(event: CustomEvent): void {
+        this.$emit('cell-action-menu-beforetoggle', event.detail);
+    }
+
+    public onActionMenuToggle(event: CustomEvent): void {
+        const detail = event.detail as MenuButtonToggleEventDetail;
```

Comment:
The CustomEvent type takes a generic for its event details, i.e. `CustomEvent<MenuButtonToggleEventDetail>`. If you do a general pass of our `CustomEvent` usage I think we can avoid a bunch of the `as` casts